### PR TITLE
Controller/PreDemo

### PR DESCRIPTION
**Pull request by alszla**
General refactor of the controller class.
Added enumeration classes to hold default values for config values, some conversions are capable of even not receiving any json at all.
Destination path value is now optional.
Miscelaneous retouches.

### CODE REVIEW

File: src/main/java/com/fundation/webservice/model/AudioConfig.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Enumeration class that holds the default key values for the config parameter within the audio transcoding endpoint
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public enum AudioConfig {
+    audioCodec("libvorbis"),
+    audioBitRate("128000"),
+    audioChannel("2"),
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/AudioConfig.java:21]: I think is better you declare like a constant. Its look magical numbers.

File: src/main/java/com/fundation/webservice/model/AudioConfig.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Enumeration class that holds the default key values for the config parameter within the audio transcoding endpoint
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public enum AudioConfig {
+    audioCodec("libvorbis"),
+    audioBitRate("128000"),
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/AudioConfig.java:20]: I think is better to declare like a constant.

File: src/main/java/com/fundation/webservice/model/AudioResponse.java [hunk 3]
```diff
@@ -11,45 +11,50 @@
...
+     * @param audioCodec      Codec use for convertion.
+     * @param audioBitRate    Bit rate of new output file
+     * @param audioChannel    number of channel for the new output file.
+     * @param metadata        defines wether the metadata will be provided as xmp or json data. If not specified by the client,
+     *                        it will not return any metadata info
      * @param checksum        information of the checksum for output file.
      */
     public AudioResponse(String fileName, String fileDownloadUri, String fileType, long size,
-                         String newFormat, String aCodec, String aBitRate, String aChannel, String checksum) {
+                         String newFormat, String audioCodec, String audioBitRate, String audioChannel, String metadata, String checksum) {
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/AudioResponse.java]: Please only 120 lines.

File: src/main/java/com/fundation/webservice/model/PdfToImageConfig.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+package com.fundation.webservice.model;
+
+/**
+ * Enumeration class that holds the default key values for the config parameter within the pdf to image conversion
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public enum PdfToImageConfig {
+    dpi("150"),
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/PdfToImageConfig.java]: Maybe I think it sould be constant it looks magical number.

File: src/main/java/com/fundation/webservice/model/VideoConfig.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Enumeration class that holds the default key values for the config parameter within the video transcoding endpoint
+ *
+ * @author Alejandro Sánchez Luizaga
+ * @version 1.0
+ */
+public enum VideoConfig {
+    audioCodec("libvorbis"),
+    audioBitRate("128000"),
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/VideoConfig.java:20]: I think the number 128000 is could be constant it looks magical number.

File: src/main/java/com/fundation/webservice/model/PdfToImageConfig.java (hunk 4 above)
Comment (JosueRodriguezGarcia) [inline src/main/java/com/fundation/webservice/model/PdfToImageConfig.java]: Change 150 -> 300

File: src/main/java/com/fundation/webservice/model/VideoConfig.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    audioChannel("2"),
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/VideoConfig.java:21]: I think the number 2 looks magical numbers.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java [hunk 7]
```diff
@@ -24,7 +24,7 @@
 	/**
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/CriteriaAudio.java:24]: Please align the comments.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java [hunk 8]
```diff
@@ -51,7 +51,7 @@ public String getAudioCodec() {
...
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/CriteriaAudio.java:51]: Align these comments.

File: src/main/java/com/fundation/webservice/model/AudioConfig.java (hunk 1 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/AudioConfig.java:21]: is a value of an ENUM Yisus, it is not a constant.

File: src/main/java/com/fundation/webservice/model/AudioConfig.java (hunk 2 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/AudioConfig.java:20]: is a value of an ENUM Yisus, it is not a constant.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java [hunk 9]
```diff
@@ -67,27 +67,27 @@ public int getAudioBitRate() {
...
+     * Method that modifies the bitrate for output file.
      *
-     * @param time The audioBitRate parameter define the number of bit rate for output file.
+     * @param audioBitRate The audioBitRate parameter define the number of bitrate for output file.
      */
     public void setAudioBitRate(int audioBitRate) {
         this.audioBitRate = audioBitRate;
     }
 
 	/**
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/CriteriaAudio.java:78]: Align the comments.

File: src/main/java/com/fundation/webservice/model/PdfToImageConfig.java (hunk 4 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/PdfToImageConfig.java]: is a value of an ENUM Yisus, it is not a constant.

File: src/main/java/com/fundation/webservice/model/VideoConfig.java (hunk 5 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/VideoConfig.java:20]: is a value of an ENUM Yisus, it is not a constant.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java [hunk 10]
```diff
@@ -67,27 +67,27 @@ public int getAudioBitRate() {
...
+     * Method that returns number of channels for output file.
      *
-     * @return The number of channel for output file.
+     * @return The number of channels for output file.
      */
     public int getAudioChannel() {
         return audioChannel;
     }
 
 	/**
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/CriteriaAudio.java:87]: Align the comments.

File: src/main/java/com/fundation/webservice/model/VideoResponse.java [hunk 11]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2019 Jalasoft.
- *
+ * <p>
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/VideoResponse.java]: Please delete this <p> p

File: src/main/java/com/fundation/webservice/model/VideoConfig.java (hunk 6 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/VideoConfig.java:21]: is a value of an ENUM Yisus, it is not a constant.

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java [hunk 12]
```diff
@@ -18,17 +19,12 @@
...
-public class KeyFrameOfVideo {
-    CriteriaKeyFrameVideo criteria;
-
-    public KeyFrameOfVideo(CriteriaKeyFrameVideo criteria) {
-        this.criteria = criteria;
-    }
-
+public class KeyFrameOfVideo implements IConvert{
     /**
      * Converts method get the keyframe depending of the number of frame.
```
Comment (LimbertVargas) [inline src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:24]: Maybe, I think to convert the format of this comment in a simple comment started with two slashes "//"

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java [hunk 13]
```diff
@@ -18,17 +19,12 @@
...
      */
-    public void convert() {
+    public void convert(CriteriaConvert criteriaConvert) {
+        CriteriaKeyFrameVideo criteria = (CriteriaKeyFrameVideo)criteriaConvert;
         try {
             String cmd = Directories.TOOLS_DIR.getDir() + Directories.FFMPEG_DIR.getDir() + "ffmpeg -y -i "
                 + criteria.getSrcPath() + " -vf fps=1/" + criteria.getTime() + " " + criteria.getDestPath()
```
Comment (TheYisusGithub) [inline src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:30]: What is fps=1 its looks magical numbers?

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java (hunk 7 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/CriteriaAudio.java:24]: There's nothing to align. This is the standard javadoc layout.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java (hunk 8 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/CriteriaAudio.java:51]: There's nothing to align. This is the standard javadoc layout.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java (hunk 9 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/CriteriaAudio.java:78]: There's nothing to align. This is the standard javadoc layout.

File: src/main/java/com/fundation/webservice/model/CriteriaAudio.java (hunk 10 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/CriteriaAudio.java:87]: There's nothing to align. This is the standard javadoc layout.

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java (hunk 12 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:24]: Only in this case, this comment is for the method. I guess this way of commenting is standard javadoc style.

File: src/main/java/com/fundation/webservice/model/AudioResponse.java (hunk 3 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/AudioResponse.java]: Fixed.

File: src/main/java/com/fundation/webservice/model/AudioConfig.java (hunk 2 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/AudioConfig.java:20]: This a enumeration class intended to define default values and constant values as any other enumeration.

File: src/main/java/com/fundation/webservice/model/AudioConfig.java (hunk 1 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/AudioConfig.java:21]: This a enumeration class intended to define default values and constant values as any other enumeration.

File: src/main/java/com/fundation/webservice/model/PdfToImageConfig.java (hunk 4 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PdfToImageConfig.java]: This a enumeration class intended to define default values and constant values as any other enumeration.

File: src/main/java/com/fundation/webservice/model/VideoConfig.java (hunk 5 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/VideoConfig.java:20]: This a enumeration class intended to define default values and constant values as any other enumeration.

File: src/main/java/com/fundation/webservice/model/PdfToImageConfig.java (hunk 4 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/PdfToImageConfig.java]: Fixed

File: src/main/java/com/fundation/webservice/model/VideoConfig.java (hunk 6 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/VideoConfig.java:21]: This a enumeration class intended to define default values and constant values as any other enumeration.

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java (hunk 13 above)
Comment (LimbertVargas) [reply src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:30]: it is not a magical number is a command defect value.

File: src/main/java/com/fundation/webservice/model/VideoResponse.java (hunk 11 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/VideoResponse.java]: Fixed

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java (hunk 12 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:24]: The other class, i.e. VideoResponse has reverted its comment blocks back to single //s.
VideoResponse class is now fixed.

File: src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java (hunk 13 above)
Comment (alszla) [reply src/main/java/com/fundation/webservice/model/KeyFrameOfVideo.java:30]: That part is a string that is being feed onto the command line to ffmpeg

**Pull request merged**